client/src/components/controls/VideoSettings.vue

```diff
@@ -0,0 +1,316 @@
+
+// Menu types enum
+enum MenuType {
+	MAIN = "main",
+	QUALITY = "quality",
+	SUBTITLE = "subtitle",
+}
+const currentMenu = ref<MenuType>(MenuType.MAIN);
+const isMenuOpen = ref<boolean>(false);
+
+const qualities = useQualities();
+const captions = useCaptions();
+
+const isQualitySupported = computed(
+	() => qualities.isQualitySupported.value && qualities.videoTracks.value.length > 0
+);
+
+const isCaptionsSupported = computed(
+	() => captions.isCaptionsSupported.value && captions.captionsTracks.value.length > 0
+);
+
+const isMenuSupported = computed(() => isQualitySupported.value || isCaptionsSupported.value);
+
+const currentSubtitleDisplay = computed(() => {
+	return captions.isCaptionsEnabled.value && captions.currentTrack.value
+		? captions.currentTrack.value
+		: null;
+});
+
+function formatQuality(quality: number): string {
+	return `${quality}p`;
+}
+
+const autoQualityDisplay = computed(() => {
+	const hasActiveQuality =
+		qualities.currentVideoTrack.value === -1 &&
+		qualities.videoTracks.value.length > 0 &&
+		qualities.currentActiveQuality.value !== null;
+
+	const currentQuality = qualities.videoTracks.value[qualities.currentActiveQuality.value!];
+	return hasActiveQuality ? `Auto (${formatQuality(currentQuality)})` : "Auto";
+});
+
+const currentQualityDisplay = computed(() => {
+	const isAutoQualitySupported = qualities.isAutoQualitySupported.value;
+	const currentTrack = qualities.currentVideoTrack.value;
+	if (isAutoQualitySupported && currentTrack === -1) {
+		return autoQualityDisplay.value;
+	}
+
+	const currentQuality = qualities.videoTracks.value[currentTrack];
+	if (currentTrack >= 0 && currentQuality) {
+		return formatQuality(currentQuality);
+	}
+
+	return "";
+});
+
+const isAutoQualityActive = computed(() => qualities.currentVideoTrack.value === -1);
+
+function isSubtitleTrackActive(track: string): boolean {
+	return captions.isCaptionsEnabled.value && track === captions.currentTrack.value;
+}
+
+function navigateToMenu(menu: MenuType): void {
+	currentMenu.value = menu;
+}
+
+function resetToMainMenu(): void {
+	currentMenu.value = MenuType.MAIN;
+}
+
+function toggleMenu(): void {
+	isMenuOpen.value = !isMenuOpen.value;
+	if (!isMenuOpen.value) {
+		resetToMainMenu();
+	}
+}
+
+function closeMenu(): void {
+	isMenuOpen.value = false;
+	resetToMainMenu();
+}
+
+function selectQuality(idx: number): void {
+	qualities.currentVideoTrack.value = idx;
+	closeMenu();
+}
+
+function selectSubtitleTrack(track: string): void {
+	if (!captions.isCaptionsEnabled.value) {
+		captions.isCaptionsEnabled.value = true;
+	}
+	captions.currentTrack.value = track;
+	closeMenu();
+}
+</script>
+
+<style lang="scss">
+@use "./media-controls.scss";
+
+.video-settings-wrapper {
+	position: relative;
+}
+
+.settings-menu-container {
+	position: absolute;
+	bottom: media-controls.$video-controls-height;
+	right: -90px;
+	z-index: 9999;
+	background: rgba(0, 0, 0, 0.7);
```

Comment:
The background shouldn't be transparent. It also doesn't respect theme colors.

This is with the strawberry theme:

<img width="367" height="256" alt="image" src="https://github.com/user-attachments/assets/f5051659-e209-4a72-9376-3a28664c4fca" />

<img width="302" height="551" alt="image" src="https://github.com/user-attachments/assets/384ded48-f846-42e4-a41a-5ae3238aa648" />